lib/index.js

```diff
@@ -101,9 +113,34 @@ export function handle (i18next, options = {}) {
 export function plugin (instance, options, next) {
   options.attachLocals = true
   const middleware = handle(options.i18next, options)
-  instance.addHook('preHandler', (request, reply, next) => middleware(request, reply, next))
+  instance.addHook('preHandler', (request, reply, next) =>
+    middleware(request, reply, next)
+  )
   return next()
 }
+
+export const hapiPlugin = {
+  name: 'i18next-http-middleware',
+  version: '1',
+  register: (server, options) => {
+    options.attachLocals = true
+    const middleware = handle(options.i18next, {
+      ...options
+    })
+
+    server.ext('onPreHandler', (request, h) => {
+      middleware(
+        request,
+        h.response() || request.Response || request.response,
```

Comment:
This line probably needs to be just `request.raw.res` this will fix the header issue

Reply:
that worked, thanks! 👍